crates/core/tedge_mapper/src/c8y/dynamic_discovery.rs

```diff
@@ -0,0 +1,83 @@
+use std::ffi::OsString;
+
+use inotify::{Event, EventMask, Inotify, WatchMask};
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum EventType {
+    ADD,
+    REMOVE,
+    UNSUPPORTED,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct DiscoverOp {
+    pub ops_dir: String,
+    pub event_type: EventType,
+    pub operation_name: String,
+}
+
+#[derive(thiserror::Error, Debug)]
+#[allow(clippy::enum_variant_names)]
+pub enum DynamicDiscoverOpsError {
+    #[error("No Operations directory path")]
+    NoOperations,
+
+    #[error("No event")]
+    NoEventName,
+
+    #[error("operation not supported")]
+    UnsupportedEvent,
+}
+
+pub fn create_inotify_stream(ops_dir: Option<String>) -> Result<Inotify, DynamicDiscoverOpsError> {
```

Comment:
why is `ops_dir` an Option? what happens when it is None?

Reply:
Will return an error, with error message saying, directory not found.